en/docs/install-and-setup/setup/security/logins-and-passwords/harshicrop-vault-extension.md

```diff
@@ -0,0 +1,298 @@
+    ```
+
+6. Then append `org-wso2-carbon-securevault-hashicorp` to the `loggers` list in the same file as follows:
+
+    ```properties
+    loggers = AUDIT_LOG, trace-messages, ..., org-wso2-carbon-securevault-hashicorp
+    ```
+
+### Step 3: Update passwords with their aliases
+
+1. Open the `deployment.toml` file in the `<APIM_HOME>/repository/conf/` directory and add
+   the `[secrets]` configuration section **at the bottom of the file** as shown below.
+   Give an alias for the passwords and put the value as blank (`""`).
+
+    ```toml
+    [secrets]
+    admin_password = ""
+    ```
+
+2. Add the encrypted password alias to the relevant sections in the `deployment.toml`
+   file by using a placeholder: `$secret{alias}`. For example:
+
+    ```toml
+    [super_admin]
+    username="admin"
+    password="$secret{admin_password}"
+    ```
+
+### Step 4: Start the server
+
+1. Provide the `VAULT_TOKEN` to the prompted message in the console or by creating a new file in the `<APIM_HOME>` 
+   directory. The file should be named according to your Operating System.
+
+    === "Linux"
+        ```
+        The file name should be `hashicorpRootToken-tmp`.
+        ```
+      
+    === "Windows"
+        ```
+        The file name should be `hashicorpRootToken-tmp.txt`.
+        ```
+
+    !!! note
+        When you add "tmp" to the file name, note that this will automatically get deleted from the file system after the server starts. Alternatively, if you want to retain the password file after the server starts, the file should be named as follows:
+
+        === "Linux"
+            ```
+            The file name should be `hashicorpRootToken-persist`.
+            ```
+        
+        === "Windows"
+            ```
+            The file name should be `hashicorpRootToken-persist.txt`.
+            ```
+
+
+2. Start the WSO2 API Manager Server and enter the keystore password at startup when prompted:
+
+    ```bash
+    [Enter KeyStore and Private Key Password :] wso2carbon
+    ```
+
+---
+
+## Setting up using AppRole authentication
+
+### Step 1: Setup HashiCorp Vault
+
+1. Start the HashiCorp Vault server and set the environment variables:
+
+    ```bash
+    export VAULT_ADDR='http://127.0.0.1:8200'
+    export VAULT_TOKEN='<root token>'
+    ```
+
+2. Include the policy in the `kv-read-write.hcl` file as below:
+
+    ```hcl
+    path "wso2apim/data/*" {
+        capabilities = ["create", "read", "update", "delete", "list"]
+    }
+    ```
+
+3. Upload the kv policy as below:
+
+    ```bash
+    vault policy write kv-read-write kv-read-write.hcl
+    ```
+
+4. The following commands can be used to create the AppRole using the created policy:
+
+    ```bash
+    vault auth enable approle
+    ```
+    ```bash   
+    vault write auth/approle/role/my-role \
+        token_policies="kv-read-write" \
+        token_type="service" \
+        token_ttl="24h" \
+        token_max_ttl="72h"
+    ```
+
+5. Get the role-id as below:
```

Comment:
```suggestion
5. Get the role-id using the command below.
```

Reply:
corrected in 1d8bdffa01d44e5da79703dd2060f0a621bb659a